HelpSource/Classes/List.schelp

```diff
@@ -6,17 +6,52 @@ categories::Collections>Ordered
 DESCRIPTION::
 List is a subclass of SequenceableCollection with unlimited growth in size. Although not a subclass of link::Classes/Array:: or its superclass link::Classes/ArrayedCollection:: it uses an Array in its implementation and is in many cases interchangeable with one. (List implements many of the same methods as Array.)
 
-Arrays have a fixed maximum size. If you add beyond that size a new Array is created and returned, but you must use an assignment statement or the new array will be lost. (See the link::Classes/Array:: helpfile.) List has no size limitation and is thus more flexible, but has slightly more overhead.
+Many of List's methods are inherited from link::Classes/SequenceableCollection:: or link::Classes/Collection:: and are documented in those helpfiles.
+
+subsection::Size limitations of arrays and flexibility of lists in SuperCollider
+
+Arrays have a fixed maximum size. Adding beyond that size, a new link::Classes/Array:: is created and returned. In order to be able to use it, the variable holding the old array can be reassigned to the new one. Threfore, the idiomatic way to add new item in link::Classes/Array:: is:
+
+code::
+x = x.add(i);
+::
+
+Example:
+
 code::
 (
-x = Array.new(3);
-y = List.new(3);
-5.do({ |i| z = x.add(i); y.add(i) });
-x.postln; z.postln; y.postln;
+// Create a new array with maxSize 3:
+x = Array.new(3); y = nil;
+"Make an empty array of size 3: \nx:% \ny: %\n".format(x, y).postln;
+
+// Try to add 5 items to the array:
+5.do { |i| y = x.add(i) };
+// If the array size exceeds maxSize, .add returns a new array .
+
+// Now, x's size is 4, which is one size bigger than the initial x, while, y is the new array with added items.
+"After adding 5 items to each: \nx:%\ny: %\n".format(x, y).postln;
+
+// Attempt to access the 5th item (index 4):
+"Accessing:\nx.at(4): \nx:%\ny: %".format(x.at(4), y[4]).postln;""
+// x.at(4) returns nil (out of bounds), y[4] returns the value (4).
 )
 ::
 
-Many of List's methods are inherited from link::Classes/SequenceableCollection:: or link::Classes/Collection:: and are documented in those helpfiles.
+List has no size limitation and is thus more flexible, but has slightly more overhead:
+code::
+(
+// Create a new List with maxSize 3 (but the size can still grow):
+x = List.new(3);
+
+// Print the empty List:
+x.postln;
+
+// Add numbers 0 to 4 to the List, printing the List after each addition:
+5.do({ |i|
+    x.add(i).postln;  // .add modifies the List in place, returns the same List.
+});
+)
+::
```

Comment:
suggestion for brevity (and clarification... List doesn't have a maxSize arg):
code::
(
// Create a new empty List with size: 3
x = List.new(3);

x.postln; // -> List[  ]

// Add numbers 0 to 4 to the List
5.do({ |i| x.add(i) });

// The list object has been modified in place with each added item
x.postln; // -> List[ 0, 1, 2, 3, 4 ]
)
::

Reply:
>List doesn't have a maxSize arg

Of course! Sorry!

Done in https://github.com/supercollider/supercollider/pull/6536/commits/027b55d840163edc880635a823f21bb4652aa5ab.

Thanks!